lib/Backend/Commands.hs

```diff
@@ -329,6 +339,70 @@ copyCmd (CopyOptions oldPath newPath force) = do
     getOperationPaths :: CopyOperation -> (EntryPath, EntryPath)
     getOperationPaths (CopyOperation old new) = (old ^. E.path, new ^. E.path)
 
+    checkForEntriesInPath :: [CopyOperation] -> Sem r ()
+    checkForEntriesInPath operations = do
+      invalidOperations <- flip filterM operations \(CopyOperation _ new) -> do
+        let path = Path.entryPathAsPath $ new ^. E.path
+        if length (path ^. pathSegments) > 1 then do
+          let pathInits = Path.EntryPath . NE.fromList <$> (init . tail . inits) (path ^. pathSegments)
+          filterM pathIsEntry pathInits >>= \case
+            [] -> pure False
+            _  -> pure True
+        else
+          pure False
+      when (not (null invalidOperations)) do
+        invalidOperations
+          <&> (\(CopyOperation old new) -> (old ^. E.path, new ^. E.path) )
+          & throw . CPRParentDirectoryIsEntry
+
+    checkForEntriesInSubpaths :: [CopyOperation] -> Sem r ()
+    checkForEntriesInSubpaths operations = do
+      invalidOperations <- flip filterM operations \(CopyOperation old new) -> do
+        checkForEntriesInSubpaths_ (Path.entryPathAsPath $ old ^. E.path) (Path.entryPathAsPath $ new ^. E.path)
+      when (not (null invalidOperations)) do
+        invalidOperations
+          <&> (\(CopyOperation old new) -> (old ^. E.path, new ^. E.path) )
+          & throw . CPRSubdirectoriesContainEntries
```

Comment:
Test `entry-directory copy conflict` covers it.

Reply:
That test expects the following error message:

> The following entries cannot be copied because a destination directory has an entry in it's subpath.

But `CPRSubdirectoriesContainEntries` prints the following error message:

> The following entries cannot be copied because a directory has an entry in it's subpath.

Reply:
Hmm. I think this check is redunant, because now we're checking `CopyOperations`, which already have all necessary entries.

Reply:
Done